### PR TITLE
Detach thread in executeAsynchronously (RuntimeExecutor.h)

### LINKED ISSUE #31088: executeAsynchronously discards constructed std::thread
https://github.com/facebook/react-native/blob/6e6443afd04a847ef23fb6254a84e48c70b45896/ReactCommon/runtimeexecutor/ReactCommon/RuntimeExecutor.h#L42-L44

the std::thread ctor is nodiscard in the latest MSVC; std::thread doesn't support fire-and-forget, in order to do that you'd have to call .detach() on it

### PR DESCRIPTION
**Pull request by asklar**
std::thread's constructor is nodiscard. This breaks in MSVC 16.9 when nodiscard starts to be enforced. Either we should hold on to the created object or detach the temporary which is what I think this function intends to do anyway.

Fixes #31088 

## Summary

Fixes an invalid usage of std::thread's constructor

## Changelog

[General] [Fixed] - fixes usage of std::thread in runtime executor

### CODE REVIEW

**Comment by asklar** [comment]
CC @TheSavior @rozele @appden

**Comment by asklar** [comment]
FYI we don't call into this function from RNW, but we get the warning/error. It's surprising that this ever works, since the standard mandates that the destructor for std::thread must terminate the program if the thread is not in a joinable state (i.e. if the thread is still running).

**Review by dannyvv — APPROVED** [review]

**Comment by asklar** [comment]
@appden any update on getting this merged?
